helpers.go

```diff
@@ -245,42 +245,57 @@ func getImageID(name string) (string, error) {
 
 // commandFunc represents a function that accepts an image ID and the CLI
 // context. This is used in the commandInContainer function.
-type commandFunc func(string, *cli.Context)
+type commandFunc func(string, *cli.Context) error
 
 // commandInContainer extracts the containerID from the given ctx. The function
 // first checks whether the container exists. If the container is running the
 // given commandFunc is executed on the image the container is based on. If the
 // --force flag is set the container is committed to a new image instead. In
 // case the container is not running it is committed to an image first.
 // Afterward commandFunc is executed on the new image.
-func commandInContainer(f commandFunc, ctx *cli.Context) {
+func commandInContainer(f commandFunc, ctx *cli.Context) (string, error) {
 	containerID := ctx.Args().First()
+	var image string
+	var err error
 	// check if the container exists
 	if !checkContainerExists(containerID) {
 		logAndFatalf("Container %s does not exist.\n", containerID)
 	}
 	// check whether the container is running
-	if container, err := checkContainerRunning(containerID); err != nil {
+	if container, runErr := checkContainerRunning(containerID); runErr != nil {
 		logAndPrintf("Checking stopped container %s ...\n", containerID)
 		// execute commitAndExecute on the stopped container
-		err = commitAndExecute(f, ctx, containerID)
-		if err != nil {
-			logAndFatalf("%v.\n", err)
-		}
+		image, err = commitAndExecute(f, ctx, containerID)
 	} else {
 		// if the force flag is set commitAndExecute is executed on the running container
 		if ctx.GlobalBool("force") {
 			logAndPrintf("WARNING: Force flag used. Manually installed packages will be analyzed as well.\n")
-			err = commitAndExecute(f, ctx, containerID)
-			if err != nil {
-				logAndFatalf("%v.\n", err)
-			}
+			image, err = commitAndExecute(f, ctx, containerID)
 		} else {
 			// directly call commandFunc on the image the container is based on
 			logAndPrintf("WARNING: Only the source image from this container will be inspected. Manually installed packages won't be taken into account.\n")
-			f(container.Image, ctx)
+			err = f(container.Image, ctx)
+			image = container.Image
+		}
+	}
+	return image, err
+}
+
+// check whether the given error has a zypper exit code != 0 and decide if it is severe
+func checkError(image, cmd string, err error) {
+	if err == nil {
+		return
+	}
+	switch err.(type) {
+	case dockerError:
+		de := int(err.(dockerError).exitCode)
```

Comment:
I am not sure if the semantics of this function are correct as is. Remember, we need to return *all* of zypper's exit codes. There is a global variable `zypperExitCode` that is used in the main function that we should set here, or am I missing something?

My proposal is to add an else branch setting `zypperExitCode`.

Reply:
zypperExitCode == dockerError.exitCode. This is set in the startContainer(...) function

Reply:
No if the exit code is severe than humanizeCommandError(..) is called and the program is terminated with exit code 1 (exitCode 1 = zypperExitErrBug). If it is not severe than the non severe exit code is return and the program terminated afterwards. 
This is why i initially wanted to remove the the exitWithCode(...) call from inside the if.

Reply:
I get, that an exit code needs to be provided. That's why the checkError (now exitOnError) function is called right before the program terminates anyway. so that now the last thing that is being done is to decide which exit code is returned.